src/super_gradients/training/models/kd_models/kd_module.py

```diff
@@ -0,0 +1,62 @@
+from super_gradients.training.models.sg_module import SgModule
+import torch
+from super_gradients.training.utils.utils import HpmStruct
+
+
+class KDOutput:
+    def __init__(self, student_output: torch.Tensor = None, teacher_output: torch.Tensor = None):
+        self.student_output = student_output
+        self.teacher_output = teacher_output
+
+
+class KDModule(SgModule):
+    """
+    KDModule
+
+    class implementing Knowledge Distillation logic as an SgModule
+
+    attributes:
+        student: torch.nn.Module - the student model
+        teacher: torch.nn.Module- the teacher model
+        freeze_teacher_eval_mode: bool- whether to freeze self.teacher at eval mode regardless of self.train(mode)
+    """
+    def __init__(self, student: torch.nn.Module, teacher: torch.nn.Module,
+                 freeze_teacher_eval_mode: bool = False):
+        super(KDModule, self).__init__()
+        self.student = student
+        self.teacher = teacher
+        self.freeze_teacher_eval_mode = freeze_teacher_eval_mode
+        self._freeze_teacher()
+
+    def _freeze_teacher(self):
```

Comment:
@shaydeci can you explain this. The teacher's params are set with require_grad=False. and if freeze_teacher_eval_mode then will be switched to train mode but still with require_grad=False?

Reply:
See above comment regarding freeze_teacher_eval_mode flag

Reply:
removed with torch no_grad()  in forward pass